domain/metadata/transaction_metadata.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from domain.metadata.metadata import Metadata, MetadataType
+
+
+@dataclass
+class MetaTransaction:
+    id: str
+    context: Optional[str]
+    genesis: Optional[bool]
+
+
+@dataclass
+class TransactionMetadata(Metadata):
+    data: Optional[MetaTransaction] = None
```

Comment:
Why are we using two classes? I feel it would be simper to use one class with all fields there. Well, we can group in subclasses for specific cases if there are too many fields.